lms/views/onedrive.py

```diff
@@ -21,8 +24,17 @@ def redirect_uri(_request):
     renderer="json",
 )
 def verify_domain(request):
-    return {
-        "associatedApplications": [
-            {"applicationId": request.registry.settings["onedrive_client_id"]}
-        ]
-    }
+    application_id = request.registry.settings["onedrive_client_id"]
+    response_text = textwrap.dedent(
+        f"""\
+    {{
+      "associatedApplications": [
+        {{
+          "applicationId": "{application_id}"
+        }}
+      ]
+    }}"""
+    )
+    request.response.content_length = len(response_text)
```

Comment:
I was wrong before, we are not sending a content type header. Now we both set it and include exactly the same json as renderer in the verification site.

Reply:
Does this line do anything? I don't think setting `request.response.*` has any effect if the view then returns a _different_ `Response` object. It'd have to be `return Response(..., content_length=len(response_text))`. But I don't think you need to do either: I think you're seeing the correct `Content-Length` because Pyramid (or WebOb) is computing it for you automatically.

The purpose of `request.response` is for when views are using a renderer like the JSON renderer or the Jinja2 renderer where the view just returns a dict (or whatever type of return value the renderer needs). In those cases the view can set `request.response.*` properties (usually headers) and the renderer will pick those up.

Reply:
Yup, I did that first, still using the json rendered and didn't look back. `Response` sets the content-length on it's own.